code/modules/mob/living/carbon/human/species_types/diona.dm

```diff
@@ -0,0 +1,371 @@
+	. = FALSE
+	var/radiation = H.radiation
+	//Dionae heal and eat radiation for a living.
+	H.adjust_nutrition(clamp(radiation, 0, 7))
+	if(radiation > 50)
+		H.heal_overall_damage(1,1, 0, BODYTYPE_ORGANIC)
+		H.adjustToxLoss(-2)
+		H.adjustOxyLoss(-1)
+
+/datum/species/diona/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+	if(chem.type == /datum/reagent/toxin/plantbgone)
+		H.adjustToxLoss(3)
+		H.reagents.remove_reagent(chem.type, chem.metabolization_rate)
+		return TRUE
+	if(chem.type == /datum/reagent/toxin/mutagen)
+		H.adjustToxLoss(-3)
+		H.reagents.remove_reagent(chem.type, chem.metabolization_rate)
+		return TRUE
+	if(chem.type == /datum/reagent/plantnutriment)
+		H.adjustBruteLoss(-1)
+		H.adjustFireLoss(-1)
+		H.reagents.remove_reagent(chem.type, chem.metabolization_rate)
+		return TRUE
+	return ..()
+
+/datum/species/diona/on_hit(obj/projectile/P, mob/living/carbon/human/H)
+	if(P.type == (/obj/projectile/energy/floramut || /obj/projectile/energy/florayield))
+		H.set_nutrition(min(H.nutrition+30, NUTRITION_LEVEL_FULL))
+
+/datum/species/diona/spec_death(gibbed, mob/living/carbon/human/H)
+	if(gibbed)
+		QDEL_NULL(H)
+		return
+	split_ability.Trigger(TRUE)
+
+/datum/species/diona/on_species_gain(mob/living/carbon/human/H)
+	. = ..()
+	var/obj/item/organ/appendix/appendix = H.getorganslot("appendix") //No appendixes for plant people
+	if(appendix)
+		appendix.Remove(H)
+		QDEL_NULL(appendix)
+	split_ability = new
+	split_ability.Grant(H)
+	partition_ability = new
+	partition_ability.Grant(H)
+	ADD_TRAIT(H, TRAIT_MOBILE, "diona")
+
+/datum/species/diona/on_species_loss(mob/living/carbon/human/H, datum/species/new_species, pref_load)
+	. = ..()
+	split_ability.Remove(H)
+	QDEL_NULL(split_ability)
+	partition_ability.Remove(H)
+	QDEL_NULL(partition_ability)
+	REMOVE_TRAIT(H, TRAIT_MOBILE, "diona")
+	for(var/status_effect as anything in H.status_effects)
+		if(status_effect == STATUS_EFFECT_PLANTHEALING)
+			H.remove_status_effect(STATUS_EFFECT_PLANTHEALING)
+
+/datum/species/diona/random_name(gender, unique, lastname, attempts)
+	. = "[pick(GLOB.diona_names)]"
+	if(unique && attempts < 10 && findname(.))
+		return ..(gender, TRUE, null, ++attempts)
+
+//////////////////////////////////////// Action abilities ///////////////////////////////////////////////
+
+/datum/action/diona/split
+	name = "Split"
+	desc = "Split into our seperate nymphs."
+	background_icon_state = "bg_default"
+	icon_icon = 'icons/mob/actions/actions_spells.dmi'
+	button_icon_state = "split"
+	var/Activated = FALSE
+
+/datum/action/diona/split/Trigger(special)
+	. = ..()
+	var/mob/living/carbon/human/user = owner
+	if(!isdiona(user))
+		return FALSE
+	if(special)
+		fakeDeath(FALSE, user) //This runs when you are dead.
+		return TRUE
+	if(user.incapacitated(ignore_restraints = TRUE)) //Are we incapacitated right now?
+		return FALSE
+	if(alert("Are we sure we wish to kill ourselves and split into seperated nymphs?",,"Yes", "No") != "Yes")
```

Comment:
```suggestion
	if(tgui_alert(usr, "Are we sure we wish to devolve ourselves and split into separated nymphs?",, list("Yes", "No")) != "Yes")
```

Reply:
I changed the wording of kill to *devolve*, as a reference to the ling ability that lets humans devolve into monkeys. (Lesser Form)

Yes, its true the old form "dies", but the term referring to being "killed" threw me off. 

You are free to change the wording I used, just keep that in mind that there's probably a better word than saying the starting mob dies.